guaranteed-message-delivery/README.md

```diff
@@ -0,0 +1,390 @@
+
+The flow is as follows.
+
+1. The publisher service receives a request with a JSON payload in the following form.
+
+    ```json
+    {
+        "patient": {
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "+94710889877",
+            "email": "johndoe@gmail.com"
+        },
+        "doctor": "thomas collins",
+        "hospital_id": "grandoaks",
+        "hospital": "grand oak community hospital",
+        "appointment_date": "2023-10-02"
+    }
+    ```
+
+2. The publisher service publishes the receiving payload to the message broker.
+
+3. The consumer service acquires the message from the message broker, extracts the necessary details (e.g. patient, doctor, hospital) and makes a call to the hospital backend service to request an appointment. A response similar to the following will be returned from the hospital backend service on success.
+
+    ```json
+    {
+        "appointmentNumber": 1,
+        "doctor": {
+            "name": "thomas collins",
+            "hospital": "grand oak community hospital",
+            "category": "surgery",
+            "availability": "9.00 a.m - 11.00 a.m",
+            "fee": 7000.0
+        },
+        "patient": {
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "8770586755",
+            "email": "johndoe@gmail.com"
+        },
+        "hospital": "grand oak community hospital",
+        "confirmed": false,
+        "appointmentDate": "2023-10-02"
+    }
+    ```
+
+4. Call the SMS service endpoint to send a message to the patient's mobile number.
+
+### Concepts covered
+
+- REST API
+- Message broker
+- HTTP Client
+- Twilio client
+
+## Develop the application
+
+### Step 1: Set up the workspace
+
+Install [Ballerina Swan Lake](https://ballerina.io/downloads/) and the [Ballerina Swan Lake VS Code extension](https://marketplace.visualstudio.com/items?itemName=wso2.ballerina) on VS Code.
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+    ```
+    $ bal new guaranteed-message-delivery
+    ```
+
+2. Remove the `main.bal` file and open the diagram view in VS Code.
+
+    ![Open diagram view](./resources/open_diagram_view.gif)
+
+3. Create a file named `types.bal` and generate record types corresponding to the payloads from the hospital backend service by providing samples of the expected JSON payload.
```

Comment:
We decided to follow the flow for new tutorials? Use the request payload in the example and then say do the same for the rest.